pkg/build/gobuild.go

```diff
@@ -349,11 +371,30 @@ func spdx(version string) sbomber {
 	}
 }
 
+func writeSBOM(sbom []byte, appPath, dir, ext string) error {
+	if dir != "" {
+		appDir := path.Join(dir, appPath)
```

Comment:
This should be `filepath.Join` in case we're running on Windows.

Should we `filepath.Clean` this too, just in case?